### DISCUSSION TITLE
Can I create records using a simple HTTP client, NO SDK?

### DISCUSSION THREAD

**Discussion by SyedMSawaid** [pocketbase/pocketbase#1129] [Q&A] [answered]
There isn't a stable C# SDK right now. I have heard one guy is working on it but I don't know how long it is going to take for him.

So can I just create records using a simple HTTP client in C#?

**Comment by ganigeorgiev** [comment, marked as answer]
> There isn't a stable C# SDK right now. I have heard one guy is working on it but I don't know how long it is going to take for him.

I'm not sure if you are referring to this specific one, but the following C# SDK was posted here before - https://github.com/PRCV1/pocketbase-csharp-sdk

---

> So can I just create records using a simple HTTP client in C#?

Yes, PocketBase provides a simple web API (the SDKs also work with it).
You can find documentation either in the "API Preview" panel for each collection in the Admin UI OR check the docs under the "Web APIs" sidebar section in https://pocketbase.io/docs/api-records/.

**Reply by SyedMSawaid** [reply]
Just saw that section. How could I miss it. 🤦🏻

**Reply by javierfuentesm** [reply]
@SyedMSawaid  maybe I'm blind but I dont see in that link how to make a request without the sdk or I dont know if the docs changed . I dont see how to authenticate the request for example or how to pass the token

**Reply by suelitonx** [reply]
@javierfuentesm  For example, when you create a collection without changing any rules (which will restrict it to superusers only), the instruction appears for you to pass the token in the request, however, if you change the listing rules (for example), the instruction disappears. I don't know if it's on purpose, but really, you're not blind.

Without changing any rules :
![Screenshot_5](https://github.com/user-attachments/assets/ac1056ff-9262-4d01-9a4a-12aaccf7f8e2)

when changing api rules:
![Screenshot_4](https://github.com/user-attachments/assets/08a033a2-4942-4d1b-9ac5-c3c742e1c4d2)


Sorry for my English, I'm using Google Translate.

**Reply by javierfuentesm** [reply]
Thank you so much!! I didn't see that! 🤦🏻‍♂️ and no worries about the English. I'm not native, neither

**Comment by dschissler** [comment]
@SyedMSawaid If you can duplicate this fetch call then you should be able to access it from anything.  You'll have to set the token as a header.  Just check a browser debug tool to see what is sent.

https://github.com/pocketbase/js-sdk/blob/6fcc1ee70b7f52108ef37d0310892ec6ebcc4fd4/src/Client.ts#L243
